### ISSUE TITLE
Adding GVNPass makes swpp-compiler yield different output with benchmark

### ISSUE THREAD

**Issue by cdh981009** [snu-sf-class/swpp202201#59]
I was planning to add some existing optimizations to the compiler for this sprint.

I only added GVN pass to opt, built the compiler, and ran some benchmark programs with the interpreter and diff the results.
And I found out that the some outputs were different from expected. Is this thing can happen?
I expect existing llvm passes to be sound and adding them to be trivial.

for example, running `binary_tree input3.txt` outputs different result with the example output in the benchmark folder.
the desired output (`binary_tree/test/output3.txt`) starts with `2 8 10..` but the compiler with GVN pass outputs `2 6 8 10..`

Also, when I run binary_tree input4.txt
`> Runtime error at ~~binary_tree/out/a.s:111: freeing non-allocated address`

Removing GVN and rebuilding the compiler makes outputs the same (as it should)

Thank you

**Comment by strikef** [comment]
This was a bug in the backend, and a fix has just been updated. We're very sorry for such frequent bugfix.

**Comment by cdh981009** [comment]
I think you forgot to exclude changes in opt.cpp? opt.cpp in the base compiler repo contains GVN pass now

**Comment by strikef** [comment]
You're right... This shouldn't have been added to the upstream. I just removed it from the upstream.
